src/framework/mocha/Utilities/MOFunctionArgument.m

```diff
@@ -1095,4 +1095,63 @@ + (NSInteger)structureToJSValue:(JSValueRef *)value inContext:(JSContextRef)ctx
     return c - c0 - 1;
 }
 
++ (BOOL)isHighSierraOrHigher {
+    NSOperatingSystemVersion version = [[NSProcessInfo processInfo] operatingSystemVersion];
+    return (version.minorVersion >= 13);
+}
+
+/**
+ This method is only called when running 10.13 or higher.
+ 
+ The method takes a symbol type string like '{CGRect="origin"{CGPoint}"size"{CGSize}}' and creates a dictionary
+ with keys = ["CGPoint", "CGSize"]. Whose values are: ['CGPoint="x"d"y"d', 'CGSize="width"d"height"d']
+ 
+ The method does this by looking up the type descriptions for each of the structs.
+ */
++ (NSDictionary <NSString *, NSString *> *)memberStructs:(NSString *)symbolType {
+    NSMutableDictionary <NSString *, NSString *> *memberStructsDictionary = [NSMutableDictionary new];
+    
+    // First strip off the initial { and final } characters from the symbol type after checking the string is long enough.
```

Comment:
I first attempted to do this when the bridging support xml is being consumed but I wasn't getting anything back when asking for the CGPoint, CGSize descriptions which is why I switched. With the other overheads associated with running the javascript I think it would be better to wait and see what the impact on performance is before adding a cache.

There is no where else that is doing quite what this code does. Because the previous bridging support supplies the full description for each struct and that is the information needed, so the name of the struct of a member is ignored, whereas that is the information we are after.

I'll add the assertion for the characters to be removed.